Sources/Brave/Frontend/Browser/TabManager.swift

```diff
@@ -416,10 +416,27 @@ class TabManager: NSObject {
     }
     // When bulk adding tabs don't notify delegates until we are done
     self.isRestoring = true
+    
+    // Load at most X of the most recent tabs
+    // IE: Load the last X tabs, lazy load all the rest
+    let amountOfTabsToRestoreImmediately = 5
+    
     var tab: Tab!
-    for url in urls {
-      tab = self.addTab(URLRequest(url: url), flushToDisk: false, zombie: zombie, isPrivate: isPrivate)
+    for (index, url) in urls.enumerated() {
+      tab = self.addTab(URLRequest(url: url),
+                            flushToDisk: false,
+                            zombie: true,
+                            isPrivate: isPrivate)
+      tab.lastTitle = url.absoluteDisplayString
+      tab.url = url
+      tab.favicon = FaviconFetcher.getIconFromCache(for: url) ?? Favicon.default
+      
+      if index >= urls.count - 1 - amountOfTabsToRestoreImmediately {
```

Comment:
This reads a bit strange, I assume this is loading the _last_ 5 items in the `urls` list? Are these already ordered correctly?

Reply:
This loads the last 5 items yeah. When we pass in the `urls` they are expected to already be in the correct order. This function only gets called from `Bookmarks` when you open all bookmarks for a folder at once.
So if a folder contains 10 bookmarks, it will open them in that exact order when you long-press on the folder and hit `Open All (10)`.

It will Lazy Load the last 5, and immediately load the first 5.
So when the user opens the tab tray, they will see the 5 pages loaded. If they scroll up, the others will not be loaded yet until they tap on them. Tabs are ordered top down in tab-tray.